lumibot/tools/polygon_helper.py

```diff
@@ -160,6 +165,45 @@ def get_price_data_from_polygon(
     return df_all
 
 
+def validate_cache(force_cache_update: bool, asset: Asset, cache_file: Path, api_key: str):
+    """
+    If the list of splits for a stock have changed then we need to invalidate its cache
+    because all of the prices will have changed (because we're using split adjusted prices).
+    Get the splits data from Polygon only once per day per stock.
+    Use the timestamp on the splits feather file to determine if we need to get the splits again.
+    When invalidating we delete the cache file and return force_cache_update=True too.
+    """
+    if asset.asset_type != Asset.AssetType.STOCK:
+        return force_cache_update
+    cached_splits = pd.DataFrame()
+    splits_file_stale = True
+    splits_file_path = Path(str(cache_file).rpartition(".feather")[0] + "_splits.feather")
+    if splits_file_path.exists():
+        splits_file_stale = datetime.fromtimestamp(splits_file_path.stat().st_mtime).date() != date.today()
+        if splits_file_stale:
+            cached_splits = pd.read_feather(splits_file_path)
+    if splits_file_stale or force_cache_update:
+        polygon_client = RESTClient(api_key)
```

Comment:
A new Polygon client is only created once per day per stock.  This is nearly the minimum number of times the client could be created.  To avoid creating another client on this call in the caller (get_price_data_from_polygon) would be super complicated because we usually don't create one here and they don't always create one up there either.

Your reviews should include more context than just the nearest lines of code in order to be helpful.  Such super local analysis should be limited to actual or likely bugs and significant anti-patterns or major readability problems.